evals/record.py

```diff
@@ -340,6 +341,46 @@ def record_final_report(self, final_report: Any):
 
         logging.info(f"Final report: {final_report}. Logged to {self.event_file_path}")
 
+class HttpRecorder(RecorderBase):
+    def __init__(self, url: str, run_spec: RunSpec):
+        super().__init__(run_spec)
+        self.url = url
+        logger.info(f"HttpRecorder initialized with URL {self.url}")
+
+    def _flush_events_internal(self, events_to_write: Sequence[Event]):
+        for event in events_to_write:
+            self._send_event(event)
```

Comment:
By default, we should probably send these events as a batch as opposed to one by one.

Reply:
Thank you for your feedback, @jwang47!
You're right, batching the events can improve the efficiency of data transfer. 

However, in this design, I wanted to maintain granular control and flexibility. That's why, for instance, I implemented batching at a different level in the evals collector. This allows for the optimization of data transmission while still maintaining control at a more granular level. You can see the implementation here: https://github.com/nstankov-bg/oaievals-collector/blob/19ee7df9f946dbc0a9769768ac3040c2f7112924/pkg/kafka/kafka.go#L38

If it's a must, I will gladly change that.

Reply:
Alvin, thank you again for the feedback.
Giving it some thought, I did:

Added a new command line argument '--http-batch-size' to allow users to specify the number of events to send in each HTTP request when running evaluations in HTTP mode. The default value is 1, which means events are sent individually.